aten/src/ATen/core/LazyTensorImpl.cpp

```diff
@@ -0,0 +1,31 @@
+#include <ATen/core/LazyTensor.h>
+
+namespace at {
+
+// We notify all siblings that they don't need to update this value
+LazyTensorImpl::~LazyTensorImpl() {
+  for (auto* sibling : siblings()) {
```

Comment:
This doesn't look thread safe.

Reply:
I'm going to reiterate an early suggestion that I made: instead of manually threading tensors together via a sibling mechanism, it seems clearer and more direct to explicitly allocate an object representing an operation, and then have tensors reference the operation directly (and only specify what their output is). Then when you move a lazy tensor from unresolved to resolved, this proceeds by updating the *operation node*, rather than the tensor.